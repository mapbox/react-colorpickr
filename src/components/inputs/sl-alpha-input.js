'use strict';

import React from 'react';
import PropTypes from 'prop-types';
import NumberInput from './number-input';

class SLAlphaInput extends React.Component {
  static propTypes = {
    theme: PropTypes.object.isRequired,
    label: PropTypes.string.isRequired,
    value: PropTypes.number.isRequired,
    onChange: PropTypes.func.isRequired
  };

  render() {
    return <NumberInput {...this.props} min={0} max={100} />;
  }
}

export default SLAlphaInput;
