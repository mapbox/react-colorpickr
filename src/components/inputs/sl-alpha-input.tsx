import React from 'react';
import { NumberInput } from './number-input.tsx';

function SLAlphaInput(props) {
  return <NumberInput {...props} min={0} max={100} />;
}

export { SLAlphaInput };
