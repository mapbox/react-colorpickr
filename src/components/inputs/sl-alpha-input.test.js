import React from 'react';
import { render } from '@testing-library/react';
import { SLAlphaInput } from './sl-alpha-input';

describe('SLAlphaInput', () => {
  test('renders', () => {
    const props = {
      id: 's',
      value: 92,
      theme: {},
      onChange: jest.fn()
    };

    const { baseElement } = render(<SLAlphaInput {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});
