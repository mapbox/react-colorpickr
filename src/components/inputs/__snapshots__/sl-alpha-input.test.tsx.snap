// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SLAlphaInput renders 1`] = `
<body>
  <div>
    <div>
      <label
        for="s"
      >
        s
      </label>
      <input
        id="s"
        max="100"
        min="0"
        step="1"
        type="number"
        value="92"
      />
    </div>
  </div>
</body>
`;
